docs/best-practices/retaining-correct-variable-types.md

```diff
@@ -1,6 +1,40 @@
 ---
 title: Retaining Correct Variable Types
 ---
+### Adding Type Annotations to Your Resolver
+`dara.core.interactivity.plain_variable.Variable`s are stored in the browser and `dara.core.interactivity.derived_variable.DerivedVariable`s and `py_component`s are calculated on the server. When a `Variable` is passed to one of these, it is serialized into a format that FastApi can handle and becomes a python basic type format on the server side. Eg, a  `numpy.array` will became a  `list`, a `numpy.int8` will become a `int`.
+
+
+By default, Dara supports to serialize all the generic data types in Numpy. See [default data types supported](https://github.com/causalens/dara/blob/master/packages/dara-core/dara/core/internal/encoder_registry.py).
```

Comment:
> By default, Dara supports to serialize all the generic data types in Numpy.

Out of the box, Dara comes with encoders for all generic data types in `pandas` and `numpy`. See...